### PR TITLE
Various syntax fixes found by FPT tool (static Fortran analysis mode)

### PR DESCRIPTION
**Pull request by JCollinsSimCon**
TYPE: New feature

KEYWORDS: Static analysis, regression testing

SOURCE: John Collins (SimCon / Edge Hill University)

CHANGES:
The WRF code has a large number of nonstandard constructs that have been in the code for a while. Part of the capability 
of the FPT static analysis tool is to discover noncompliant Fortran syntax and usage. 
1. A few issues in the existing WRF code preclude the FPT utility from running on WRF, including outright Fortran bugs. 
The bugs are removed. 
2. Additionally, the temporary files in the frame directory associated with the automatic code generation of namelist 
configuration processing are no longer removed by the Makefile. This was not a bug, but the static analyzer needs
the cpp manufactured code to continue to exist.

This PR represents an initial step. Once the WRF code is suitable for automatic use by FPT, then a second PR will 
introduce the required exemplar files for comparison. That step, which insures no additional nonstandard Fortran 
constructs or usages are introduced, will eventually be part of the jenkins testing.

MODIFIED FILES:
dyn_em/module_first_rk_step_part1.F
   * Inserted missing continuation
   * The leading "&" character in column 6 on the following line was sufficient for GNU, Intel, PGI

external/io_grib2/bacio-1.3/bacio.F
   * Inserted missing "::" delimiters
   * a declaration with an assignment requires the "::" delimeters

frame/Makefile
* Commented-out deletion of xx*.f90 and yy*.f90.    
* These are needed for the regression test and are deleted by the `clean` script

phys/module_bl_mynn.F
   * Additional + sign deleted at line 3030

TESTS CONDUCTED:  
1. fpt regression tests
2. Jenkins is all PASS.

RELEASE NOTE: Changes to implement static analysis regression testing of WRF.  Please see:
http://simconglobal.com/WRF_Workshop_June_2011_Poster_Automatic_Detection_of_Software_Errors_in_WRF.pdf
http://simconglobal.com/WRF_Workshop_June_2012_Poster_QA_Analysis_of_the_WRF_Program.pdf
http://simconglobal.com/collins_et_al_2013_automated_quality_assurance_analysis_wrf_a_case_study.pdf

### CODE REVIEW

File: frame/Makefile [hunk 1]
```diff
@@ -89,82 +90,82 @@ $(LIBTARGET) :  $(MODULES) $(OBJS) $(SPECIAL) $(NLOBJS) $(ALOBJS)
 nl_set_0_routines.o : nl_access_routines.F module_configure.o
 	$(CPP) -DNNN=0 -I../inc -DNL_set_ROUTINES nl_access_routines.F > xx0.f90
 	$(FC) -o $@ -c $(PROMOTION) $(FCNOOPT) $(FCBASEOPTS_NO_G) $(MODULE_DIRS) $(FCSUFFIX) xx0.f90
-	$(RM) xx0.f90
+#	$(RM) xx0.f90
```
Comment (davegill) [inline frame/Makefile:93]: @JCollinsSimCon 
John
Maybe we should just delete all of these lines

**Comment by JCollinsSimCon** [comment]
@davegill 
@brianfarrimond
@ellagrp
@nikosspanakis
2 more files to go for WRF_fpt_regression

**Comment by davegill** [comment]
@JCollinsSimCon 
John,
Likely you received an email from "jenkins" (it might be in your spam folder). Would you please copy the text of that email into one of the comment blocks at the bottom of the PR: https://github.com/wrf-model/WRF/pull/1253

This validates that your code did no irreparable harm to the repository, as far as we can tell.

**Comment by JCollinsSimCon** [comment]
Please find result of the WRF regression test cases in the attachment. This build is for Commit ID: 6cf16beebf68701be4666835f0a3d31215393b47, requested by: JCollinsSimCon for PR: https://github.com/wrf-model/WRF/pull/1253. For any query please send e-mail to David Gill.

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 19           18
    Number of Builds       : 48           46
    Number of Simulations  : 166           164        0
    Number of Comparisons  : 105           104        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Comment by JCollinsSimCon** [comment]
@davegill 
Assuming that module_fr_fire_phys.F has already been taken care of, I believe we have all of the files we need for WRF_fpt_regression - dyn_em/module_first_rk_step_part1.F, frame/Makefile, external/io_grib2/bacio-1.3/bacio.F, phys/module_bl_mynn.F.  Am I right?

**Comment by davegill** [comment]
John,
Thanks, I'll look soon
Dave

File: frame/Makefile [hunk 2]
```diff
@@ -1,4 +1,5 @@
-#
+# Makefile for /frame
```
Comment (davegill) [inline frame/Makefile:1]: @JCollinsSimCon 
John,
Remove these leading lines

**Comment by davegill** [comment]
@joeolson42 @weiwangncar @dudhia 
Joe,
Would you please give your approval to this mod to your code. There is a syntax error with a double plus sign. I know that GNU, PGI, and Intel allow this, but it is a nonstandard coding. The fix makes no changes to the results.

**Comment by davegill** [comment]
@JCollinsSimCon 
John,
I am good with the PR now (I eliminated the commented-out "rm" commands). We are waiting for a developer to chime in on his thumbs up to the scheme.

You may now stand down.

**Pull request merged**